apps/full-site-editing/full-site-editing-plugin/full-site-editing/plugins/template-update-notice/index.js

```diff
@@ -0,0 +1,28 @@
+/* global fullSiteEditing */
+/**
+ * External dependencies
+ */
+import domReady from '@wordpress/dom-ready';
+import { render } from '@wordpress/element';
+import { Notice } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+domReady( () => {
+	if ( 'wp_template_part' !== fullSiteEditing.editorPostType ) {
+		return;
+	}
+	const element = document.createElement( 'div' );
+	element.classList.add( 'wp-block-a8c-template-edit-warning' );
```

Comment:
Just curious, could we use `dispatch('core/notices').createNotice( ... )` here? Maybe something along the following lines:

```javascript
wp.data.dispatch( 'core/notices' ).createNotice(
    'warning',
    'Updates to this block will affect multiple pages on your site.',
    { isDismissible: false }
);
```

Reply:
ah, good point - my vision was narrowed here by the way the custom publish button was done - dispatching a notice is a much less hacky way so I will give that a try - thanks.

Reply:
That worked and is a much nicer solution so have gone with that.